tools/scylla-nodetool.cc

```diff
@@ -72,6 +72,64 @@ using tools::utils::typed_option;
 using scylla_nodetool_app_template = tools::utils::tool_app_template<scylla_rest_client&, const bpo::variables_map&>;
 using operation = scylla_nodetool_app_template::operation;
 
+enum class json_type {
+    null, boolean, object, array, string, number
+};
+
+const rjson::value& check_json_type(const rjson::value& value, json_type type) {
+    bool ok = false;
+    sstring type_name = "unknown";
+    switch (type) {
+        case json_type::null:
+            ok = value.IsNull();
+            type_name = "null";
+            break;
+        case json_type::boolean:
+            ok = value.IsBool();
+            type_name = "bool";
+            break;
+        case json_type::object:
+            ok = value.IsObject();
+            type_name = "object";
+            break;
+        case json_type::array:
+            ok = value.IsArray();
+            type_name = "array";
+            break;
+        case json_type::string:
+            ok = value.IsString();
+            type_name = "string";
+            break;
+        case json_type::number:
+            ok = value.IsNumber();
+            type_name = "number";
+            break;
+        default:
+            throw std::runtime_error(fmt::format("check_json_type(): uknown type: {}", static_cast<int>(type)));
+    }
+    if (!ok) {
+        throw std::runtime_error(fmt::format("check_json_type(): json value is not of the expected type: {}", type_name));
+    }
+    return value;
+}
+
+void compact_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (vm.count("keyspace")) {
+        std::unordered_map<sstring, sstring> params;
+        const auto keyspace = vm["keyspace"].as<sstring>();
+        if (vm.count("tables")) {
+            params["cf"] = fmt::to_string(fmt::join(vm["tables"].as<std::vector<sstring>>(), ","));
+        }
+        client.post(format("/storage_service/keyspace_compaction/{}", keyspace), std::move(params));
+    } else {
+        auto keyspaces_json = client.get("/storage_service/keyspaces", {{"type", "all"}});
+        for (const auto& keyspace_json : check_json_type(keyspaces_json, json_type::array).GetArray()) {
+            const auto keyspace = rjson::to_string_view(check_json_type(keyspace_json, json_type::string));
+            client.post(format("/storage_service/keyspace_compaction/{}", keyspace));
```

Comment:
Is this what the regular nodetool/jmx thing does?

Reply:
According to my experimenting with the command itself, yes. I should also check the code I suppose.

Reply:
I checked the code and this is what nodetool does too.